tests/PersonDialogTest.java

```diff
@@ -0,0 +1,204 @@
+        // into the respective boxes
+        dialog.textBox("firstName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_U, VK_F, VK_F, VK_I, VK_N);
+        dialog.textBox("lastName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_A, VK_N);
+        dialog.textBox("address").pressAndReleaseKeys(VK_1, VK_2, VK_3, VK_4, VK_SPACE).pressKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_D).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_R, VK_E, VK_A, VK_R, VK_Y, VK_SPACE).pressKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_L).releaseKey(VK_SHIFT).pressAndReleaseKeys(VK_A, VK_N, VK_E);
+        dialog.textBox("city").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_C).releaseKey(VK_SHIFT)
+                .pressAndReleaseKeys(VK_O, VK_O, VK_K, VK_I, VK_E, VK_SPACE).pressKey(VK_SHIFT).pressAndReleaseKeys(VK_C)
+                .releaseKey(VK_SHIFT).pressAndReleaseKeys(VK_I, VK_T, VK_Y);
+        dialog.textBox("state").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_W, VK_A).releaseKey(VK_SHIFT);
+        dialog.textBox("zip").pressAndReleaseKeys(VK_6, VK_6, VK_6, VK_6, VK_6);
+        dialog.textBox("phone").pressAndReleaseKeys(VK_1, VK_2, VK_3, VK_4, VK_5, VK_6, VK_7, VK_8, VK_9, VK_0);
+
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is added
+        window.table().requireRowCount(1);
+
+    }
+
+    /**
+     * This tests that a person cannot be added with improper characters.
+     */
+
+    @Test
+    void addPersonInvalid() {
+        // Test when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        dialog.textBox("firstName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_1).releaseKey(VK_SHIFT)
+            .pressAndReleaseKeys(VK_2, VK_2, VK_4, VK_5, VK_6);
+        
+        dialog.textBox("lastName").pressKey(VK_SHIFT).pressAndReleaseKeys(VK_M).releaseKey(VK_SHIFT)
+            .pressAndReleaseKeys(VK_9, VK_7);
+        
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is not added
+        window.table().requireRowCount(0);
+
+    }
+
+
+    @Test
+    void addPersonNullFirst() {
+        // Test when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
+
+        dialog.textBox("firstName").pressKey(VK_SPACE);
+
+        // Click 'OK'
+        dialog.button(JButtonMatcher.withText("OK")).click();
+
+        // Test person is not added
+        window.table().requireRowCount(0);
+
+    }
+
+
+    @Test
+    void addPersonMixedInvlaid() {
+        // Test first name valid and lastname invalid when user exits with OK
+        window.button("add").click();
+        DialogFixture dialog = window.dialog();
```

Comment:
So if you just remove the AddressBookGUI part, you can skip all these clicks to get to the PersonDialog

Reply:
Whenever I try to remove the AddressBookGUI part, I get an EDT violation. I am still trying a different way around it though.